com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/Raytracing/HDRaytracingAmbientOcclusion.cs

```diff
@@ -59,103 +54,279 @@ static RTHandle AmbientOcclusionHistoryBufferAllocatorFunction(string viewName,
+        {
+            AmbientOcclusionComposeParameters aoComposeParameters = new AmbientOcclusionComposeParameters();
+            var aoSettings = hdCamera.volumeStack.GetComponent<AmbientOcclusion>();
+            aoComposeParameters.validRayTracing = m_RenderPipeline.GetRayTracingState();
+            aoComposeParameters.intensity = aoSettings.intensity.value;
+            aoComposeParameters.actualWidth = hdCamera.actualWidth;
+            aoComposeParameters.actualHeight = hdCamera.actualHeight;
+            aoComposeParameters.viewCount = hdCamera.viewCount;
+            aoComposeParameters.aoShaderCS = m_PipelineRayTracingResources.aoRaytracingCS;
+            aoComposeParameters.intensityKernel = m_RTAOApplyIntensityKernel;
+            return aoComposeParameters;
+        }
+
+        AmbientOcclusionComposeResources PrepareAmbientOcclusionComposeResources(HDCamera hdCamera, RTHandle inOutTexture)
+        {
+            AmbientOcclusionComposeResources aoComposeResources = new AmbientOcclusionComposeResources();
+            aoComposeResources.outputTexture = inOutTexture;
+            return aoComposeResources;
+        }
+
+        public void RenderRTAO(HDCamera hdCamera, CommandBuffer cmd, RTHandle outputTexture, ShaderVariablesRaytracing globalCB, ScriptableRenderContext renderContext, int frameCount)
+        {
+            AmbientOcclusionTraceParameters aoTraceParameters = PrepareAmbientOcclusionTraceParameters(hdCamera, globalCB);
+            AmbientOcclusionTraceResources aoTraceResources = PrepareAmbientOcclusionTraceResources(hdCamera, m_AOIntermediateBuffer0);
+            TraceAO(cmd, aoTraceParameters, aoTraceResources);
+
+            AmbientOcclusionDenoiseParameters aoDenoiseParameters = PrepareAmbientOcclusionDenoiseParameters(hdCamera, globalCB);
+            AmbientOcclusionDenoiseResources aoDenoiserResources = PrepareAmbientOcclusionDenoiseResources(hdCamera, m_AOIntermediateBuffer0, m_AOIntermediateBuffer1, outputTexture);
+            DenoiseAO(cmd, hdCamera, aoDenoiseParameters, aoDenoiserResources);
+
+            AmbientOcclusionComposeParameters aoComposeParameters = PrepareAmbientOcclusionComposeParameters(hdCamera, globalCB);
+            AmbientOcclusionComposeResources aoComposeResources = PrepareAmbientOcclusionComposeResources(hdCamera, outputTexture);
+            ComposeAO(cmd, hdCamera, aoComposeParameters, aoComposeResources);
+        }
+
+        static public void TraceAO(CommandBuffer cmd, AmbientOcclusionTraceParameters aoTraceParameters, AmbientOcclusionTraceResources aoTraceResources)
         {
             // If any of the previous requirements is missing, the effect is not requested or no acceleration structure, set the default one and leave right away
-            if (!m_RenderPipeline.GetRayTracingState())
+            if (!aoTraceParameters.validRayTracing)
```

Comment:
This should be outside of the static function.
In render graph, default fallbacks are handled differently so the code needs to be outside in order not to interfere.